torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -280,6 +282,110 @@ def softmax_bwd_rule(op_schema: OpSchema) -> OutputSharding:
+    assert len(op_schema.args_schema) == 5
+    (
+        input_strategy,
+        target_strategy,
+        weight_strategy,
+        reduction,
+        _,
+    ) = op_schema.args_schema
+    input_strategy = cast(OpStrategy, input_strategy)
+    target_strategy = cast(OpStrategy, target_strategy)
+    reduction = cast(int, reduction)
+
+    input_shape = input_strategy.output_shape
+    channel_dim = 1 if len(input_shape) >= 2 else 0
+
+    output_strategy = OpStrategy([])
+    for idx, input_placement_strategy in enumerate(input_strategy.strategies):
+        op_args_target_specs = []
+        redistribute_costs = []
+
+        # make sure input is replicated along the channel dim
+        input_src_spec = input_placement_strategy.output_spec
+        input_tgt_spec = DTensorSpec(
+            mesh=mesh,
+            placements=replicate_reduction_dims(
+                input_src_spec.placements, [channel_dim]
+            ),
+            tensor_meta=input_src_spec.tensor_meta,
+        )
+        op_args_target_specs.append(input_tgt_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(input_strategy, input_tgt_spec)
+        )
+
+        # target doesn't have channel dim, and it follows input on other dims
+        target_placement_strategy = target_strategy.strategies[idx]
+        target_src_spec = target_placement_strategy.output_spec
+        target_tgt_spec = DTensorSpec(
+            mesh=mesh,
+            placements=_skip_dim(input_tgt_spec.placements, channel_dim),
+            tensor_meta=target_src_spec.tensor_meta,
+        )
+        op_args_target_specs.append(target_tgt_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(target_strategy, target_tgt_spec)
+        )
+
+        # weight tensor, if given, has to be a Tensor of size input_shape[channel_dim]
+        # make sure it is replicated
+        if weight_strategy is not None:
+            assert isinstance(weight_strategy, OpStrategy)
+            weight_src_spec = weight_strategy.strategies[idx].output_spec
+            weight_tgt_spec = DTensorSpec(
+                mesh=mesh,
+                placements=_replicate_dims_start_at(weight_src_spec.placements),
+                tensor_meta=weight_src_spec.tensor_meta,
+            )
+            op_args_target_specs.append(weight_tgt_spec)
+            redistribute_costs.append(
+                generate_redistribute_costs(weight_strategy, weight_tgt_spec)
+            )
+
+        if reduction == Reduction.NONE.value:
+            output_tgt_spec = target_tgt_spec
+        else:
+            if reduction == Reduction.MEAN.value:
+                reduction_op = c10d.ReduceOp.AVG
+                if not is_tensor_evenly_shardable(
+                    target_tgt_spec.shape, target_tgt_spec
+                ):
+                    raise ValueError(
+                        "The intermediate results of nll_loss cannot be evenly sharded, \
```

Comment:
> "The intermediate results" 

what intermediate results referring to here?

Reply:
It's referring to the results with `reduction='none'`. When `reduction='mean'`, there is an extra step to collapse the big tensor into a scalar.